lib/basket.rb

```diff
@@ -0,0 +1,67 @@
+require 'csv'
+
+class Basket
+  def initialize(customer_order)
+    @customer_order = customer_order
+    @basket_data = []
+  end
+
+  def show_basket
+    store_menu_data
+    populate_basket_data
+    header
+    basket_breakdown
+    footer
+  end
+
+  private
+
+  def header
+    puts "BASKET"
+    puts "=" * 39
+  end
+
+  def footer
+    puts "-" * 39
+    puts "TOTAL:".ljust(32, ' ') + "£%.2f" % basket_total
+  end
+
+  def basket_total
+    @basket_data.reduce(0) { |sum, order| sum + order[:subtotal] }
+  end
+
+  def basket_breakdown
+    @basket_data.each do |order|
+      puts "#{order[:quantity]}x".ljust(5, ' ') +
+           "#{capitalise_item(order[:dish])}(s)".ljust(26, ' ') +
+           " £#{order[:subtotal]}"
+    end
+  end
+
+  def populate_basket_data
+    @customer_order.each do |dish, quantity|
+      @basket_data << { dish: dish, quantity: quantity, subtotal: sub_total(dish, quantity) }
+    end
+  end
+
+  def sub_total(dish, quantity)
+    (@menu_data[dish].to_f * quantity).round(2)
+  end
+
+  def capitalise_item(item)
+    item.split.map { |word| word.capitalize }.join(" ")
+  end
+
+  def store_menu_data
+    @menu_data = load_data('../menu.csv')
```

Comment:
My understanding is that the reason the code is structured this way is to avoid many instances of the same menu existing. What if the `initialize` method for this class accepted an instance of Menu as input? The reason this might be a better approach is that imagine someone creates a menu, makes an order, and then a dev accidentally edits this csv. This means that the view of the menu items will be inconsistent between the Order and Menu. I also expect that this will reduce some repeated code between your classes, as each currently needs to load and process the csv data